### DISCUSSION TITLE
For how long will `pages` directory be supported?

### DISCUSSION THREAD

**Discussion by nikmace** [vercel/next.js#44678] [Help] [answered]
We would like to use _some_ of the new features NextJS 13 offers, but we are _unable to migrate_ to the `app` directory because it breaks our app and it is too expensive to switch. From the blog, I can see that it will continue to be supported in the _foreseeable future_. But what does that mean? If we switch to NextJS 13 and you will stop supporting it, we won't be able to continue the project.

Therefore, the question is will the`pages` directory continue to be supported in NextJS 13 (and until when), and when it will stop being supported?

**Comment by icyJoseph** [comment, marked as answer]
While there is not an exact word on the time, I do believe the page directory will be supported for several major versions. That being said, it feels like it was just yesterday that Next 10 was announced, and we are already at 13.1.x - 

My guess is that it'll be there  for~ever~ some years, receiving support and updates.

**Reply by tbscode** [reply]
I really hope you will be able to support accessing the request object with the new app directive in nexjs 13.

https://github.com/vercel/next.js/discussions/42732

Otherwise this will break a whole array of use cases for nextjs, especially more comprehensive server setups.
